ddtrace/contrib/dbapi/__init__.py

```diff
@@ -19,67 +19,49 @@
         if not pin or not pin.enabled():
-            return self.__wrapped__.executemany(query, *args, **kwargs)
+            return method(*args, **kwargs)
         service = pin.service
 
-        # FIXME[matt] properly handle kwargs here. arg names can be different
-        # with different libs.
-        with pin.tracer.trace(self._datadog_name, service=service, resource=query) as s:
+        with pin.tracer.trace(self._self_datadog_name, service=service, resource=resource) as s:
             s.span_type = sql.TYPE
-            s.set_tag(sql.QUERY, query)
+            s.set_tag(sql.QUERY, resource)
             s.set_tags(pin.tags)
-            s.set_tag("sql.executemany", "true")
+            s.set_tags(extra_tags)
+
             try:
-                return self.__wrapped__.executemany(query, *args, **kwargs)
+                return method(*args, **kwargs)
             finally:
                 s.set_metric("db.rowcount", self.rowcount)
 
-    def execute(self, query, *args, **kwargs):
-        pin = self._datadog_pin
-        if not pin or not pin.enabled():
-            return self.__wrapped__.execute(query, *args, **kwargs)
+    def executemany(self, query, *args, **kwargs):
+        # FIXME[matt] properly handle kwargs here. arg names can be different
+        # with different libs.
+        return self._trace_method(
+            self.__wrapped__.executemany, query, {'sql.executemany': 'true'},
+            query, *args, **kwargs)
 
-        service = pin.service
-        with pin.tracer.trace(self._datadog_name, service=service, resource=query) as s:
-            s.span_type = sql.TYPE
-            s.set_tag(sql.QUERY, query)
-            s.set_tags(pin.tags)
-            try:
-                return self.__wrapped__.execute(query, *args, **kwargs)
-            finally:
-                s.set_metric("db.rowcount", self.rowcount)
+    def execute(self, query, *args, **kwargs):
+        return self._trace_method(
+            self.__wrapped__.execute, query, {}, query, *args, **kwargs)
 
     def callproc(self, proc, args):
-        pin = self._datadog_pin
-        if not pin or not pin.enabled():
-            return self.__wrapped__.callproc(proc, args)
-
-        with pin.tracer.trace(self._datadog_name, service=pin.service, resource=proc) as s:
-            s.span_type = sql.TYPE
-            s.set_tag(sql.QUERY, proc)
-            s.set_tags(pin.tags)
-            try:
-                return self.__wrapped__.callproc(proc, args)
-            finally:
-                s.set_metric("db.rowcount", self.rowcount)
+        return self._trace_method(self.__wrapped__.callproc, proc, {}, proc,
+                                  args)
 
     def __enter__(self):
         # previous versions of the dbapi didn't support context managers. let's
         # reference the func that would be called to ensure that errors
         # messages will be the same.
-        self.__wrapped__.__enter__
+        self.__wrapped__.__enter__()
```

Comment:
this needs to be called? I'm missing a bit this part since it was one of our first integrations, but I think here we just wanted to raise an error if the context manager wasn't available. What test is failing / working now after this change?